src/main/java/com/minecolonies/colony/buildings/AbstractBuilding.java

```diff
@@ -11,50 +16,47 @@
 import com.minecolonies.tileentities.TileEntityColonyBuilding;
 import com.minecolonies.util.BlockPosUtil;
 import com.minecolonies.util.Log;
+
 import io.netty.buffer.ByteBuf;
 import net.minecraft.block.Block;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.tileentity.TileEntity;
 import net.minecraft.util.BlockPos;
 import net.minecraftforge.fml.common.gameevent.TickEvent;
 
-import java.lang.reflect.Constructor;
-import java.lang.reflect.InvocationTargetException;
-import java.util.HashMap;
-import java.util.Map;
-
 /**
- * Base building class, has all the foundation for what a building stores and does.
+ * Base building class, has all the foundation for what a building stores and
+ * does.
  */
 public abstract class AbstractBuilding
 {
-    private         final       BlockPos                    location;
-    private         final       Colony                      colony;
+    private final BlockPos location;
+    private final Colony   colony;
 
     private MaterialStore materialStore;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Private field 'materialStore' could be made final; it is only initialized in the declaration or constructor. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AImmutableField)